processing/src/main/java/io/druid/segment/StringDimensionIndexer.java

```diff
@@ -415,23 +413,15 @@ public IndexedInts getRow()
         if (indices == null || indices.length == 0) {
           final int nullId = getEncodedValue(null, false);
           if (nullId > -1) {
-            if (nullId < maxId) {
-              row = new int[] {nullId};
-              rowSize = 1;
-            } else {
-              // Choose to use ArrayBasedIndexedInts later, instead of EmptyIndexedInts, for monomorphism
-              row = IntArrays.EMPTY_ARRAY;
-              rowSize = 0;
-            }
+            row = new int[] {nullId};
```

Comment:
`ArrayBasedIndexedInts` could be memoized here, instead of creating new array and new ArrayBasedIndexedInts every time.